language-server/dm/rawDocument.ml

```diff
@@ -92,15 +92,15 @@ let word_at_position raw pos : string option =
     let start_ind = loc_of_position raw pos in
     (* Search backwards until we find a character that cannot be part of a word *)
     let first_non_word_ind = Str.search_backward back_reg raw.text start_ind in
+    let first_word_ind = first_non_word_ind + 1 in
     let forward_reg = Str.regexp {|[^a-zA-Z_0-9']|} in
     (* Search forwards ensuring that all characters are part of a well defined word. (Cannot start with [0-9'.] and cannot end with .)*)
     let last_word_ind = Str.search_forward forward_reg raw.text start_ind in
-    let word = String.sub raw.text first_non_word_ind (last_word_ind - first_non_word_ind) in
+    (* we get the substring from the first word index to the last index for the word *)
+    let word = String.sub raw.text first_word_ind (last_word_ind - first_word_ind) in
     Some word
-  with Not_found ->
+  with _ ->
```

Comment:
Not sure if this is idiomatic in Ocaml, but I noticed when looking at #731 that if we attempt to hover on a non-ascii character it could cause a crash due to `String.sub` raising `Invalid_argument`

Reply:
More generally I wonder (I am not really experienced in Ocaml) what the proper way for doing something akin to Regex in UTF would be to help eventually solve 731.

Reply:
Not sure how to handle this. Maybe @gares has pointers ?